common/pinot/pinotClient.go

```diff
@@ -175,20 +165,47 @@ func (c *PinotClient) convertSearchResultToVisibilityRecord(hit []interface{}, c
 		TaskList:         source.TaskList,
 		IsCron:           source.IsCron,
 		NumClusters:      source.NumClusters,
-		SearchAttributes: attr,
+		SearchAttributes: customKeyMap,
 	}
 	if source.UpdateTime != 0 {
 		record.UpdateTime = time.UnixMilli(source.UpdateTime)
 	}
 	if source.CloseTime != 0 {
 		record.CloseTime = time.UnixMilli(source.CloseTime)
-		record.Status = thrift.ToWorkflowExecutionCloseStatus(&source.CloseStatus)
+		record.Status = toWorkflowExecutionCloseStatus(source.CloseStatus)
 		record.HistoryLength = source.HistoryLength
 	}
 
 	return record
 }
 
+func toWorkflowExecutionCloseStatus(status int) *types.WorkflowExecutionCloseStatus {
+	if status < 0 {
+		return nil
+	}
+	switch status {
```

Comment:
Can we direclty cast using WorkflowExecutionCloseStatus(status), for example https://stackoverflow.com/questions/46974446/convert-an-integer-to-an-enum-value